module/VuFind/src/VuFind/View/Helper/Root/Matomo.php

```diff
@@ -0,0 +1,678 @@
+        $code .= $this->getTrackSearchCode($results, $customData);
+        $code .= $this->getClosingTrackingCode();
+
+        return $code;
+    }
+
+    /**
+     * Track a Combined Search
+     *
+     * @param Results $results         Search Results
+     * @param array   $combinedResults Combined Search Results
+     *
+     * @return string Tracking Code
+     */
+    protected function trackCombinedSearch(Results $results, array $combinedResults
+    ): string {
+        $customData = 'lightbox' === $this->context
+            ? $this->getLightboxCustomData()
+            : $this->getSearchCustomData($results);
+
+        $code = $this->getOpeningTrackingCode();
+        $code .= $this->getCustomVarsCode($customData);
+        $code .= $this->getTrackCombinedSearchCode(
+            $results,
+            $combinedResults,
+            $customData
+        );
+        $code .= $this->getClosingTrackingCode();
+
+        return $code;
+    }
+
+    /**
+     * Track a Record View
+     *
+     * @param RecordDriverBase $recordDriver Record Driver
+     *
+     * @return string Tracking Code
+     */
+    protected function trackRecordPage(RecordDriverBase $recordDriver): string
+    {
+        $customData = 'lightbox' === $this->context
+            ? $this->getLightboxCustomData()
+            : $this->getRecordPageCustomData($recordDriver);
+
+        $code = $this->getOpeningTrackingCode();
+        $code .= $this->getCustomVarsCode($customData);
+        $code .= $this->getTrackPageViewCode($customData);
+        $code .= $this->getClosingTrackingCode();
+
+        return $code;
+    }
+
+    /**
+     * Track a Generic Page View
+     *
+     * @return string Tracking Code
+     */
+    protected function trackPageView(): string
+    {
+        $customData = 'lightbox' === $this->context
+            ? $this->getLightboxCustomData()
+            : $this->getGenericCustomData();
+
+        $code = $this->getOpeningTrackingCode();
+        $code .= $this->getCustomVarsCode($customData);
+        $code .= $this->getTrackPageViewCode($customData);
+        $code .= $this->getClosingTrackingCode();
+
+        return $code;
+    }
+
+    /**
+     * Get Search Results if on a Results Page
+     *
+     * @return ?Results Search results or null if not on a search page
+     */
+    protected function getSearchResults(): ?Results
+    {
+        $viewModel = $this->getView()->plugin('view_model');
+        $current = $viewModel->getCurrent();
+        if (null === $current || 'layout/lightbox' === $current->getTemplate()) {
+            return null;
+        }
+        $children = $current->getChildren();
+        if (isset($children[0])) {
+            $template = $children[0]->getTemplate();
+            if (!strstr($template, '/home') && !strstr($template, 'facet-list')) {
+                $results = $children[0]->getVariable('results');
+                if (is_a($results, 'VuFind\Search\Base\Results')) {
```

Comment:
I realize that this probably just crept in from the existing Piwik helper, but I'd suggest changing `is_a()` calls to `instanceof` so that we can actually refer to the class directly in the code instead of using its name in a string. That improves static analysis and allows advanced IDEs to work better.

Reply:
Forgot to mention that I changed this.

Reply:
Thanks!